lib/sequel/extensions/activerecord_connection.rb

```diff
@@ -73,6 +75,23 @@ def synchronize(*)
 
     private
 
+    def keep_transaction_options(options)
+      options.slice(
+        :auto_savepoint,
+        :isolation,
+        :num_retries,
+        :before_retry,
+        :prepare,
+        :retry_on,
+        :rollback,
+        :server,
+        :savepoint,
+        :deferrable,
+        :read_only,
+        :synchronous,
+      )
+    end
```

Comment:
Could you explain this change? I'm not sure I understand what it does and how it's related to this change 🤔

Reply:
Models' methods merge additional options to opts, not related to transaction execution, for example: https://github.com/jeremyevans/sequel/blob/master/lib/sequel/model/base.rb#L1482

I suppose there should be only transaction-related options https://github.com/jeremyevans/sequel/blob/master/lib/sequel/database/transactions.rb#L130

Reply:
add test to cover it

Reply:
Oh, right, I see what you mean. In that case, thanks for adding the corresponding integration tests, that's definitely super important 👍